src/javascript/app_2/Stores/Modules/Contract/Helpers/__tests__/logic.js

```diff
@@ -0,0 +1,362 @@
+                "profit": 100,
+            };
+            expect(Logic.getDisplayStatus(contract_info)).to.eql('won');
+        });
+        it('should return lost if contract is ended and profit is less than zero', () => {
+            const contract_info = {
+                "status": "closed",
+                "profit": -100,
+            };
+            expect(Logic.getDisplayStatus(contract_info)).to.eql('loss');
+        });
+        it('should return won if contract is ended and profit is zero', () => {
+            const contract_info = {
+                "status": "closed",
+                "profit": 0,
+            };
+            expect(Logic.getDisplayStatus(contract_info)).to.eql('won');
+        });
+        it('should return purchased if contract is not ended', () => {
+            const contract_info = {
+                "status": "open",
+            };
+            expect(Logic.getDisplayStatus(contract_info)).to.eql('purchased');
+        });
+    });
+
+    describe('getEndSpot', () => {
+        it('should return contract\'s sell spot if contract is path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": true,
+                "sell_spot": 123456,
+                "exit_tick": 987654321,
+            };
+            expect(Logic.getEndSpot(contract_info)).to.eql(123456);
+        });
+        it('should return contract\'s exit tick if contract is not path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": false,
+                "sell_spot": 123456,
+                "exit_tick": 987654321,
+            };
+            expect(Logic.getEndSpot(contract_info)).to.eql(987654321);
+        });
+        it('should return contract\'s exit tick if is_path_dependent is undefined', () => {
+            const contract_info = {
+                "sell_spot": 123456,
+                "exit_tick": 987654321,
+            };
+            expect(Logic.getEndSpot(contract_info)).to.eql(987654321);
+        });
+    });
+
+    describe('getEndSpotTime', () => {
+        it('should return contract\'s sell spot time if it is path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": true,
+                "sell_spot_time": 123456,
+                "exit_tick_time": 987654321,
+            };
+            expect(Logic.getEndSpotTime(contract_info)).to.eql(123456);
+        });
+        it('should return contract\'s exit tick time if it is not path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": false,
+                "sell_spot_time": 123456,
+                "exit_tick_time": 987654321,
+            };
+            expect(Logic.getEndSpotTime(contract_info)).to.eql(987654321);
+        });
+        it('should return contract\'s exit tick time if is_path_dependent is undefined', () => {
+            const contract_info = {
+                "is_path_dependent": false,
```

Comment:
since it's testing when `is_path_dependent` is `undefined`, could you remove this line?